### ISSUE TITLE
[Youtube] Missing formats on a video

### ISSUE THREAD

**Issue by pitsi** [yt-dlp/yt-dlp#10911]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Greece

### Provide a description that is worded well enough to be understood

I am using mpv to play a youtube playlist and this mpv profile which plays the audio only on format 140 (medium quality m4a) and shuffle it.
```
[ytm]
shuffle=yes
ytdl-format=140
display-tags-clr
```
The playlist has 20 video entries and 19 of them play just fine. The one that fails, pops this error in the terminal
```
[ytdl_hook] ERROR: [youtube] NyVHCu54ul8: Requested format is not available. Use --list-formats for a list of available formats
[ytdl_hook] youtube-dl failed: unexpected error occurred
Failed to recognize file format.
```
Upon further inspection, as you will see below, I found out that format 140 and a lot of others are not available for this video anymore! It happens both on the latest stable and on the latest nightly. 
But does it happen for anyone else or on a different video? 

p.s. Yt-dlp is NOT broken on youtube, but I am forced to check the checkmark on "I am reporting that yt-dlp is broken on a supported site" for github to let me press the "submit new issue button".

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
$ yt-dlp -vUF https://www.youtube.com/watch?v=NyVHCu54ul8
[debug] Command-line config: ['-vUF', 'https://www.youtube.com/watch?v=NyVHCu54ul8']
[debug] User config "/home/user/.config/yt-dlp/config": ['--format-sort', 'vcodec:h264,res:720,acodec:m4a', '--no-cache-dir']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.08.06 from yt-dlp/yt-dlp [4d9231208] (zip)
[debug] Python 3.12.5 (CPython x86_64 64bit) - Linux-6.10.6-amd64-x86_64-with-glibc2.40 (OpenSSL 3.3.1 4 Jun 2024, glibc 2.40)
[debug] exe versions: ffmpeg 7.0.2 (fdk,setts), ffprobe 7.0.2
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.06.02, mutagen-1.46.0, requests-2.32.3, sqlite3-3.46.0, urllib3-2.0.7, websockets-10.4
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests
[debug] Loaded 1830 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.08.06 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.08.06 from yt-dlp/yt-dlp)
[youtube] Extracting URL: https://www.youtube.com/watch?v=NyVHCu54ul8
[youtube] NyVHCu54ul8: Downloading webpage
[youtube] NyVHCu54ul8: Downloading ios player API JSON
[youtube] NyVHCu54ul8: Downloading web creator player API JSON
[debug] [youtube] Extracting signature function js_57c75fa4_111
[youtube] NyVHCu54ul8: Downloading player 57c75fa4
[debug] [youtube] Decrypted nsig Cu5yWWIA6mfcEsw => BBoYoAuBu4H3KA
[debug] Sort order given by user: vcodec:h264, res:720, acodec:m4a
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, vcodec:h264(7), res:720(720.0), acodec:m4a(9), quality, fps, hdr:12(7), source, channels, lang, proto, size, br, asr, vext, aext, hasaud, id
[info] Available formats for NyVHCu54ul8:
ID  EXT   RESOLUTION FPS CH │    FILESIZE   TBR PROTO │ VCODEC        VBR ACODEC      ABR ASR MORE INFO
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
sb3 mhtml 48x27        1    │                   mhtml │ images                                storyboard
sb2 mhtml 80x45        1    │                   mhtml │ images                                storyboard
sb1 mhtml 160x90       1    │                   mhtml │ images                                storyboard
sb0 mhtml 320x180      1    │                   mhtml │ images                                storyboard
251 webm  audio only      2 │     2.09MiB  131k https │ audio only        opus       131k 48k [el] medium, WEB-C, webm_dash
243 webm  640x360     24    │     3.39MiB  212k https │ vp9          212k video only          360p, WEB-C, webm_dash
160 mp4   256x144     24    │  1000.47KiB   61k https │ avc1.4D400C   61k video only          144p, IOS, mp4_dash
134 mp4   640x360     24    │     4.40MiB  275k https │ avc1.4D401E  275k video only          360p, IOS, mp4_dash
18  mp4   640x360     24  2 │     5.90MiB  369k https │ avc1.42001E       mp4a.40.2       44k [el] 360p, IOS
136 mp4   1280x720    24    │    16.97MiB 1061k https │ avc1.64001F 1061k video only          720p, IOS, mp4_dash

$ ./yt-dlp -vUF https://www.youtube.com/watch?v=NyVHCu54ul8
[debug] Command-line config: ['-vUF', 'https://www.youtube.com/watch?v=NyVHCu54ul8']
[debug] User config "/home/user/.config/yt-dlp/config": ['--format-sort', 'vcodec:h264,res:720,acodec:m4a', '--no-cache-dir']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2024.09.02.232855 from yt-dlp/yt-dlp-nightly-builds [e8e6a982a] (zip)
[debug] Python 3.12.5 (CPython x86_64 64bit) - Linux-6.10.6-amd64-x86_64-with-glibc2.40 (OpenSSL 3.3.1 4 Jun 2024, glibc 2.40)
[debug] exe versions: ffmpeg 7.0.2 (fdk,setts), ffprobe 7.0.2
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.06.02, mutagen-1.46.0, requests-2.32.3, sqlite3-3.46.0, urllib3-2.0.7, websockets-10.4
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests
[debug] Loaded 1832 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-nightly-builds/releases/latest
Latest version: nightly@2024.09.02.232855 from yt-dlp/yt-dlp-nightly-builds
yt-dlp is up to date (nightly@2024.09.02.232855 from yt-dlp/yt-dlp-nightly-builds)
[youtube] Extracting URL: https://www.youtube.com/watch?v=NyVHCu54ul8
[youtube] NyVHCu54ul8: Downloading webpage
[youtube] NyVHCu54ul8: Downloading ios player API JSON
[youtube] NyVHCu54ul8: Downloading web creator player API JSON
[debug] [youtube] Extracting signature function js_57c75fa4_111
[youtube] NyVHCu54ul8: Downloading player 57c75fa4
[debug] [youtube] Decrypted nsig I0L9uq12B-QBjyh => 790AbGEL2Q4ciw
[debug] Sort order given by user: vcodec:h264, res:720, acodec:m4a
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, vcodec:h264(7), res:720(720.0), acodec:m4a(9), quality, fps, hdr:12(7), source, channels, lang, proto, size, br, asr, vext, aext, hasaud, id
[info] Available formats for NyVHCu54ul8:
ID  EXT   RESOLUTION FPS CH │    FILESIZE   TBR PROTO │ VCODEC        VBR ACODEC      ABR ASR MORE INFO
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
sb3 mhtml 48x27        1    │                   mhtml │ images                                storyboard
sb2 mhtml 80x45        1    │                   mhtml │ images                                storyboard
sb1 mhtml 160x90       1    │                   mhtml │ images                                storyboard
sb0 mhtml 320x180      1    │                   mhtml │ images                                storyboard
251 webm  audio only      2 │     2.09MiB  131k https │ audio only        opus       131k 48k [el] medium, IOS, webm_dash
243 webm  640x360     24    │     3.39MiB  212k https │ vp9          212k video only          360p, WEB-C, webm_dash
160 mp4   256x144     24    │  1000.47KiB   61k https │ avc1.4D400C   61k video only          144p, IOS, mp4_dash
134 mp4   640x360     24    │     4.40MiB  275k https │ avc1.4D401E  275k video only          360p, IOS, mp4_dash
18  mp4   640x360     24  2 │     5.90MiB  369k https │ avc1.42001E       mp4a.40.2       44k [el] 360p, IOS
136 mp4   1280x720    24    │    16.97MiB 1061k https │ avc1.64001F 1061k video only          720p, IOS, mp4_dash
```

**Comment by seproDev** [comment]
This video seems to just not have format 140 available.
If you can get the AAC audio to play in your browser (right click > stats for nerds), this would be a bug in yt-dlp. But given that even with `--extractor-arg "youtube:player_client=all"` the AAC formats are not available, this seems to be a specific issue YouTube has with this video.

**Comment by jamesdengtm** [comment]
[youtube] Extracting URL: https://www.youtube.com/watch?v=AJw16T1kLf4
[youtube] AJw16T1kLf4: Downloading webpage
[youtube] AJw16T1kLf4: Downloading ios player API JSON
[youtube] AJw16T1kLf4: Downloading web creator player API JSON
ERROR: [youtube] AJw16T1kLf4: Sign in to confirm you’re not a bot. This helps protect our community. Learn more
  File "yt_dlp\extractor\common.py", line 740, in extract
  File "yt_dlp\extractor\youtube.py", line 4358, in _real_extract
  File "yt_dlp\extractor\common.py", line 1263, in raise_no_formats

**Comment by pitsi** [comment]
I know. But it did have it as available e.g. last week (I listen to that same playlist almost every day), because I noticed the issue just yesterday. And it is not just format 140 that is missing, because any other video from the same list has literally every format available
```
$ yt-dlp -F https://www.youtube.com/watch?v=5gnsNP3tWFo
[youtube] Extracting URL: https://www.youtube.com/watch?v=5gnsNP3tWFo
[youtube] 5gnsNP3tWFo: Downloading webpage
[youtube] 5gnsNP3tWFo: Downloading ios player API JSON
[youtube] 5gnsNP3tWFo: Downloading web creator player API JSON
[youtube] 5gnsNP3tWFo: Downloading player 57c75fa4
[youtube] 5gnsNP3tWFo: Downloading m3u8 information
[info] Available formats for 5gnsNP3tWFo:
ID  EXT   RESOLUTION FPS CH │   FILESIZE  TBR PROTO │ VCODEC         VBR ACODEC      ABR ASR MORE INFO
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────
sb2 mhtml 48x27        0    │                 mhtml │ images                                 storyboard
sb1 mhtml 79x45        1    │                 mhtml │ images                                 storyboard
sb0 mhtml 159x90       1    │                 mhtml │ images                                 storyboard
249 webm  audio only      2 │    1.39MiB  48k https │ audio only         opus        48k 48k [ro] low, webm_dash
250 webm  audio only      2 │    1.84MiB  64k https │ audio only         opus        64k 48k [ro] low, webm_dash
251 webm  audio only      2 │    3.68MiB 127k https │ audio only         opus       127k 48k [ro] medium, webm_dash
233 mp4   audio only        │                 m3u8  │ audio only         unknown             [ro] Default
234 mp4   audio only        │                 m3u8  │ audio only         unknown             [ro] Default
139 m4a   audio only      2 │    1.41MiB  49k https │ audio only         mp4a.40.5   49k 22k [ro] low, m4a_dash
140 m4a   audio only      2 │    3.74MiB 129k https │ audio only         mp4a.40.2  129k 44k [ro] medium, m4a_dash
602 mp4   256x144     12    │ ~  2.27MiB  79k m3u8  │ vp09.00.10.08  79k video only
603 mp4   256x144     24    │ ~  3.30MiB 114k m3u8  │ vp09.00.11.08 114k video only
278 webm  256x144     24    │    1.16MiB  40k https │ vp09.00.11.08  40k video only          144p, webm_dash
604 mp4   426x240     24    │ ~  5.03MiB 174k m3u8  │ vp09.00.20.08 174k video only
242 webm  426x240     24    │    2.22MiB  77k https │ vp09.00.20.08  77k video only          240p, webm_dash
605 mp4   640x360     24    │ ~ 11.73MiB 407k m3u8  │ vp09.00.21.08 407k video only
243 webm  640x360     24    │    4.94MiB 171k https │ vp09.00.21.08 171k video only          360p, webm_dash
606 mp4   854x480     24    │ ~ 18.27MiB 633k m3u8  │ vp09.00.30.08 633k video only
244 webm  854x480     24    │    7.08MiB 245k https │ vp09.00.30.08 245k video only          480p, webm_dash
609 mp4   1280x720    24    │ ~ 20.62MiB 715k m3u8  │ vp09.00.31.08 715k video only
247 webm  1280x720    24    │   12.10MiB 419k https │ vp09.00.31.08 419k video only          720p, webm_dash
269 mp4   256x144     24    │ ~  2.55MiB  88k m3u8  │ avc1.4D400C    88k video only
160 mp4   256x144     24    │  657.55KiB  22k https │ avc1.4D400C    22k video only          144p, mp4_dash
229 mp4   426x240     24    │ ~  3.68MiB 128k m3u8  │ avc1.4D4015   128k video only
133 mp4   426x240     24    │    1.32MiB  46k https │ avc1.4D4015    46k video only          240p, mp4_dash
230 mp4   640x360     24    │ ~  7.81MiB 271k m3u8  │ avc1.4D401E   271k video only
134 mp4   640x360     24    │    2.40MiB  83k https │ avc1.4D401E    83k video only          360p, mp4_dash
18  mp4   640x360     24  2 │ ≈  6.12MiB 212k https │ avc1.42001E        mp4a.40.2       44k [ro] 360p
231 mp4   854x480     24    │ ~  9.89MiB 343k m3u8  │ avc1.4D401E   343k video only
135 mp4   854x480     24    │    3.77MiB 130k https │ avc1.4D401E   130k video only          480p, mp4_dash
232 mp4   1280x720    24    │ ~ 22.75MiB 789k m3u8  │ avc1.4D401F   789k video only
136 mp4   1280x720    24    │   10.90MiB 378k https │ avc1.4D401F   378k video only          720p, mp4_dash
```
Even without resolutions above 720p, that is 30+ formats vs 10 for the one I am talking about! 

And the same limited number of formats is available in youtube music
```
$ yt-dlp -F https://music.youtube.com/watch?v=NyVHCu54ul8
[youtube] Extracting URL: https://music.youtube.com/watch?v=NyVHCu54ul8
[youtube] NyVHCu54ul8: Downloading webpage
[youtube] NyVHCu54ul8: Downloading ios player API JSON
[youtube] NyVHCu54ul8: Downloading web creator player API JSON
[youtube] NyVHCu54ul8: Downloading ios music player API JSON
[youtube] NyVHCu54ul8: Downloading web music client config
[youtube] NyVHCu54ul8: Downloading player 57c75fa4
[youtube] NyVHCu54ul8: Downloading web music player API JSON
[info] Available formats for NyVHCu54ul8:
ID  EXT   RESOLUTION FPS CH │    FILESIZE   TBR PROTO │ VCODEC        VBR ACODEC      ABR ASR MORE INFO
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
sb3 mhtml 48x27        1    │                   mhtml │ images                                storyboard
sb2 mhtml 80x45        1    │                   mhtml │ images                                storyboard
sb1 mhtml 160x90       1    │                   mhtml │ images                                storyboard
sb0 mhtml 320x180      1    │                   mhtml │ images                                storyboard
251 webm  audio only      2 │     2.09MiB  131k https │ audio only        opus       131k 48k [el] medium, webm_dash
243 webm  640x360     24    │     3.39MiB  212k https │ vp9          212k video only          360p, webm_dash
160 mp4   256x144     24    │  1000.47KiB   61k https │ avc1.4D400C   61k video only          144p, mp4_dash
134 mp4   640x360     24    │     4.40MiB  275k https │ avc1.4D401E  275k video only          360p, mp4_dash
18  mp4   640x360     24  2 │     5.90MiB  369k https │ avc1.42001E       mp4a.40.2       44k [el] 360p
136 mp4   1280x720    24    │    16.97MiB 1061k https │ avc1.64001F 1061k video only          720p, mp4_dash
```
I will try a few other playlists the following days and report back if I can find any other video that has the same issue.

**Comment by seproDev** [comment]
We cannot change what formats YouTube provides.
YouTube is known to sometimes remove formats.
For what reason? No idea.

**Comment by pitsi** [comment]
I know that. But when format 22 was removed in late June, it was "not available only for a few video" and now it is not avaliable for all videos. Let's hope we will not face the same.

And I am telling that as an angry youtube user on kodi who is forced to use their garbage api for more than 5 years now and put up with the issues the dash streams have. An hour ago I had to stop the ~1 hour video I was watching there and start it over with "format 18" because "format 231" started buffering like 1 minute or so after 30+ minutes of playback.
